test/functional/ex_cmds/cd_spec.lua

```diff
@@ -0,0 +1,95 @@
+-- Specs for :cd, :tcd, :lcd
+
+local helpers = require('test.functional.helpers')
+local nvim, execute, eq, clear, eval, feed =
+  helpers.nvim, helpers.execute, helpers.eq, helpers.clear, helpers.eval, helpers.feed
+
+describe(':cd :tcd', function()
+  before_each(clear)
+
+  --it('sets to local directory for the program, tabs and windows', function()
+    local wd = function()
+      return eval('getcwd()')
+    end
+
+    -- Test both spellings: cd and chdir
+    -- Their variants differ in prefix (none, 't' and 'l')
+    for _, cmd in ipairs({'cd', 'chdir'}) do
+      describe('*' .. cmd, function()
+        it('works', function()
+          -- Store the initial working directory
+          local globalDir = wd()
+
+          -- Make tree of directories
+          local directories = {
+            {name = 'lua_test_dir_1', children =  {
+                {name = 'dir_a', children = nil},
```

Comment:
`children = nil` is *absolutely* the same thing as absense of this key. Lua tables do not allow storing `nil` values.